libdevcore/RLP.cpp

```diff
@@ -149,7 +149,7 @@ unsigned RLP::length() const
 		return n - c_rlpDataImmLenStart;
 	else if (n < c_rlpListStart)
 	{
-		if ((int)m_data.size() <= n - c_rlpDataIndLenZero)
+		if ((int)m_data.size() <= n - c_rlpDataIndLenZero || m_data[1] == 0)
```

Comment:
1. we should not rely on this
2. In my case, m_data.size() = 1, n - c_rlpDataIndLenZero = 184 - 183 = 1. so the first statement is true. So it will evaluate m_data[1] and fail